### PR TITLE
Generate tile metadata for MapSets with raster children

### LINKED ISSUE #4934: Allow tilemetadata route to generate mosiacs for ScannedMaps with Raster grandchildren
https://github.com/pulibrary/figgy/blob/main/app/controllers/tile_metadata_controller.rb#L7

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #4934

### CODE REVIEW

File: app/controllers/tile_metadata_controller.rb [hunk 1]
```diff
@@ -19,6 +19,16 @@ def metadata
 
   private
 
+    def generate_mosaic?
+      if resource.is_a?(RasterResource) && resource.decorate.raster_set?
+        true
+      elsif resource.is_a?(ScannedMap) && resource.decorate.map_set?
```
Comment (hackartisan) [inline app/controllers/tile_metadata_controller.rb:25]: Won't this give a false positive for map sets that don't have any raster children?

**Review by hackartisan — CHANGES_REQUESTED** [review]

File: spec/factories/scanned_map.rb [hunk 2]
```diff
@@ -67,5 +67,21 @@
     factory :complete_scanned_map do
       state "complete"
     end
+
+    factory :scanned_map_with_raster_children do
```
Comment (tpendragon) [inline spec/factories/scanned_map.rb:71]: :heart:

File: app/controllers/tile_metadata_controller.rb (hunk 1 above)
Comment (tpendragon) [reply app/controllers/tile_metadata_controller.rb:25]: Also can we just generate this for all scanned maps/rasters? If it's a scanned map with no raster children will something blow up?

Comment (eliotjordan) [reply app/controllers/tile_metadata_controller.rb:25]: It won't blow up, just returns an error. To me, seems better than running the mosaic file_set query multiple times per request. We could rescue that error and return a 404.

Comment (eliotjordan) [reply app/controllers/tile_metadata_controller.rb]: Fixed

**Review by hackartisan — APPROVED** [review]

File: app/controllers/tile_metadata_controller.rb [hunk 3]
```diff
@@ -1,19 +1,18 @@
 # frozen_string_literal: true
 class TileMetadataController < ApplicationController
   attr_reader :resource
+  rescue_from MosaicService::Error, with: :not_found
 
   def metadata
     @resource = find_resource(params[:id])
-    if resource.is_a?(RasterResource) && resource.decorate.raster_set?
+    if resource.is_a?(RasterResource) || resource.is_a?(ScannedMap)
```
Comment (hackartisan) [inline app/controllers/tile_metadata_controller.rb:8]: Let's add a code comment here saying why we purposefully include things we know will not have a mosaic service.

**Review by hackartisan — CHANGES_REQUESTED** [review]

**Review by hackartisan — APPROVED** [review]

**Pull request merged**